lib/identity_cache/query_api.rb

```diff
@@ -400,16 +406,18 @@ def prefetch_one_association(association, records)
     def fetch_recursively_cached_association(ivar_name, association_name) # :nodoc:
       if IdentityCache.should_use_cache?
         ivar_full_name = :"@#{ivar_name}"
-
         assoc = if instance_variable_defined?(ivar_full_name)
           instance_variable_get(ivar_full_name)
         else
           instance_variable_set(ivar_full_name, send(association_name))
         end
-
-        assoc.is_a?(ActiveRecord::Associations::CollectionAssociation) ? assoc.reader : assoc
+        if IdentityCache.fetch_returns_relation
+          assoc.is_a?(ActiveRecord::Associations::CollectionAssociation) ? assoc.reader : assoc
+        else
+          assoc.to_a
```

Comment:
This method is used for cache_has_one's fetch association method as well, so we can't always call `.to_a` if `IdentityCache.fetch_returns_relation` is false